core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider.go

```diff
@@ -114,30 +123,31 @@ func (provider *GitPackageContentProvider) GetOnDiskAbsoluteFilePath(absoluteFil
 	}
 
 	// Otherwise clone the repo and return the absolute path of the requested file
-	interpretationError := provider.atomicClone(parsedURL)
-	if interpretationError != nil {
+	if interpretationError := provider.atomicClone(parsedURL); interpretationError != nil {
 		return "", interpretationError
 	}
 
-	// check whether kurtosis yaml exists in the path
-	maybeKurtosisYamlPath, interpretationError := getKurtosisYamlPathForFileUrl(pathToFileOnDisk, provider.packagesDir)
-	if interpretationError != nil {
-		return "", startosis_errors.WrapWithInterpretationError(err, "Error occurred while verifying whether '%v' belongs to a Kurtosis package.", absoluteFileLocator)
-	}
+	if shouldOnlyAcceptsPackageFilePath {
```

Comment:
i wonder if we can check if this is not true and eject immediately with `return pathTofileOnDisk, nil` and then do the checks if it is true

saves some indentation, is in line with ejecting early

Reply:
great suggestion Keep doing it @h4ck3rk3y